_sips/minutes/2020-03-11-minutes.md

```diff
@@ -0,0 +1,220 @@
+    * Gui: on the contrary, the status quo (having to add a `case`) forces you to learn this detail early
+    * Seb: but we're dropping auto-tupling?
+    * Martin: it's related to the pattern bindings changes (below)
+* Pattern bindings
+    * Martin `for (case Foo(a, b) <- ...)` for filtering, and val pattern, same
+    * https://dotty.epfl.ch/docs/reference/changed-features/pattern-bindings.html
+* Toplevel definitions
+    * Gui: details around `private`. top-level `private type` and `private class`, the type is in a synthetic object, the class is really top level
+    * that also impacts opaque types
+    * Seb: then let's force users to put in object and export instead
+    * but, boilerplate...
+* Trait parameters
+    * Gui/Seb: yes, everyone wants them
+    * Gui: something about mixing in multiple traits and them having implicit parameters (that need hand-wiring?)
+* Type lambdas
+    * Gui/Seb: yes, everyone wants them
+    * Gui: something about variance
+* Drop type projection (over abstract types)
+    * Gui: it's unsound
+    * Martin: Under `-strict` (i.e. how Scala 3.1 will behave by default) it's a compile error.  Will migrate from `-strict` to `-Xsource:3.1`
+    * Gui: we might change it so it's just lower-bonded, not upper-bounded
+    * Martin: actually, let's put that change under a `-Y` option
+    * Seb: Seems to me match types are the replacement for this
+* Weak conformance -> harmonization
+    * Seb: with overloading weak conformance is back
+    * Seb: Long and Any overloading, nsc conforms Int to Long, dotc picks Any
+    * Seb: JUnit assertEquals overloads, and Integer/Long aren't equals
+    * Seb: two parts to this change: the List case (inference) and overloading when primitives are present
+* Wildcard types
+    * `?` instead of `_`
+
+### Review enums
+
+Martin: most seem happy with the current design decisions
+
+two pushbacks:
+1. should apply return the precise type?
+2. support nested enums
+
+Seb: these conflict.  If `apply` returns the wide type, then enums can never have nested enums.
+
+Most of the committee don't feel strongly for hierachical enums ("nested" enums).
+
+Disallow extending java.lang.Enum outside of enums.
+
+### Review creator applications
+
+Seb: what's the point? Martin: avoid having to explain the difference
+
+Aka "optional `new`".
+
+### Review `@infix` and `@alpha`
+
+* `@infix`: you are allowed and encouraged to call infix
+* `@alpha`: very different.  2 things:
+
+1. it gives the platform encoded (simple) name of the method
+2. canonical name of the method (it's "Googleable" name)
+
+But you can't invoke that alpha name.  It's just for reference, like documentation.  (So why not just use scaladoc?)
+
+`@alpha` is trying to address the problem of overuse of symbolic methods.
+
+Martin: so, 2 questions:
+
+* should it be manditory?
+* is it useful?
+
+Most believe it to be useful, and seems noone is against it being manditory.
+
```

Comment:
add:
`@infix` is uncontroversial, but there is a question of whether to allow non-`@infix`-annotated methods to still be used infix before a curly brace, e.g. `xs map { ...`. conclusion: allow that in 3.0, maybe crack down harder later.